### DISCUSSION TITLE
Reverse Engineering a Protocol

### DISCUSSION THREAD

**Discussion by sk8board** [pascallanger/DIY-Multiprotocol-TX-Module#1000] [Q&A]
@pascallanger Is there a tutorial of how to reverse engineer an RC 2.4GHz protocol? 

Do you start with a logic analyzer between the microprocessor and the RF chip or do you use something like Universal Radio Hacker?

Thank you,

**Comment by pascallanger** [comment]
I've never written a tutorial...
Spi bus spying was very popular in the past and still on some models.
But most of the chips now have the RF on board. In this case you need to listen over the air with SDR and see if you can understand what's being sent.
Let me know if you have a real thing you want to reverse and I will guide you.

**Comment by sk8board** [comment]
Thank you for your offer. What SDR device do you recommend?

-------- Original Message --------

**Reply by pascallanger** [reply]
I have a Pluto SDR but a hackrf is a good choice.
